tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -581,6 +583,14 @@ internal static CreateAsyncEndMethodResult CreateAsyncEndMethodDelegate(Type int
              *      - TReturn OnAsyncMethodEnd<TTarget, TReturn>(TReturn returnValue, Exception exception, in CallTargetState state);
              *      - [Type] OnAsyncMethodEnd<TTarget>([Type] returnValue, Exception exception, in CallTargetState state);
              *
+             * Or as a Task<> return
+             *      - Task<TReturn> OnAsyncMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, CallTargetState state);
+             *      - Task<TReturn> OnAsyncMethodEnd<TTarget, TReturn>(TReturn returnValue, Exception exception, CallTargetState state);
+             *      - Task<[Type]> OnAsyncMethodEnd<TTarget>([Type] returnValue, Exception exception, CallTargetState state);
+             *      - Task<TReturn> OnAsyncMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state);
+             *      - Task<TReturn> OnAsyncMethodEnd<TTarget, TReturn>(TReturn returnValue, Exception exception, in CallTargetState state);
+             *      - Task<[Type]> OnAsyncMethodEnd<TTarget>([Type] returnValue, Exception exception, in CallTargetState state);
```

Comment:
The only case to use `Task` is when you want to use `await` as well right? Maybe we should add that here just to be explicit, otherwise people might think you _have_ to use `Task` when instrumenting async methods? And if so, we can't use `in` in the async method, so there's only 3 possible candidates?`

Also, we should add this to the integration docs


```suggestion
             *      - async Task<TReturn> OnAsyncMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, CallTargetState state);
             *      - async Task<TReturn> OnAsyncMethodEnd<TTarget, TReturn>(TReturn returnValue, Exception exception, CallTargetState state);
             *      - async Task<[Type]> OnAsyncMethodEnd<TTarget>([Type] returnValue, Exception exception, CallTargetState state);
```

Reply:
Oh you're right!